cpu/atxmega/include/cpu_ebi.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_atxmega
+ * @{
+ *
+ * @file
+ * @brief           External Bus Interface API
+ *
+ * @author          Gerson Fernando Budke <nandojve@gmail.com>
+ */
+
+#include "periph_cpu.h"
+
+#ifndef CPU_EBI_H
+#define CPU_EBI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef uint32_t hugemem_ptr_t;
+
+#define HUGEMEM_NULL        0
+
+void ebi_init(void);
+
+static inline uint_fast8_t hugemem_read8(const hugemem_ptr_t from)
+{
+    uint8_t value;
+
+    __asm__ volatile (
+        "movw r30, %A1                  \n\t"
+        "out %2, %C1                    \n\t"
+        "ld %0, Z                       \n\t"
+        "out %2, __zero_reg__           \n\t"
+        :
+        "=r"(value)
+        :
+        "r"(from), "i"(&RAMPZ)
+        :
+        "r30", "r31"
+    );
+
+    return value;
+}
+static inline void hugemem_write8(hugemem_ptr_t to, uint_fast8_t val)
+{
+    __asm__ volatile (
+        "movw r30, %A0                  \n\t"
+        "out %2, %C0                    \n\t"
+        "st Z, %1                       \n\t"
+        "out %2, __zero_reg__           \n\t"
+        :
+        :
+        "r"(to), "r"(val), "i"(&RAMPZ)
+        :
+        "r30", "r31"
+    );
+}
+uint_fast16_t hugemem_read16(const hugemem_ptr_t from);
+void hugemem_write16(hugemem_ptr_t to, uint_fast16_t val);
+uint_fast32_t hugemem_read32(const hugemem_ptr_t from);
+void hugemem_write32(hugemem_ptr_t to, uint_fast32_t val);
```

Comment:
Is there much benefit in implementing them individually? Why not just use

```C
static inline uint16_t hugemem_read16(const hugemem_ptr_t from)
{
    uint16_t result;
    hugemem_read_block(&result, from, sizeof(result));
    return result;
}
```

I'd personally would do so even for read8 and write8, so that there is less ASM to maintain.

Reply:
Agreed, by docs `LD/ST` _load/store indirect from/to data space to a register. For parts with SRAM, the data space consists of the Register File, I/O memory, and internal SRAM (and external SRAM if applicable)._

I give a try and for some unknown reason it is not working.

If this request is not a blocking, I do prefer keep this way for now and try revisit later.